### PR TITLE
Create and test populate board method

### PR DESCRIPTION
**Pull request by rjdellecese**
I disabled Rubocop in app/models/game.rb because it was giving me an error that I didn't think was a concern; the error is noted below the Rubocop disable comment in that same file.

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -1,5 +1,25 @@
-# Model for each chess game
+# rubocop:disable all
```
Comment (lasley) [inline app/models/game.rb]: I dunno if I'd outright disable Rubocop. Maybe just the specific metric? 
```
# rubocop:disable Metrics/AbcSize
```

File: app/models/game.rb [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+#  ^^^
 class Game < ActiveRecord::Base
   has_many :chess_pieces
   has_many :users
+
+  delegate :pawns, :rooks, :knights, :bishops, :queens, :kings, to: :chess_pieces
+
+  after_create :populate_board!
+
+  def populate_board!
```
Comment (lasley) [inline app/models/game.rb]: Might be worth making this private in order to avoid accidental calling

File: spec/models/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe Game, type: :model do
```
Comment (lasley) [inline spec/models/game_spec.rb]: Thorough tests :+1:

**Comment by lasley** [comment]
Looks excellent. Few notes, but nothing major :+1:

**Comment by lasley** [comment]
Whoops I meant to describe that Rubocop error in normal terms. It's basically saying that you have variable assignments nested deep in recursion or logic.

In a model it is possible for this to happen under normal circumstances, but you should still try your best to refactor around it. Typically this is a sign that your code is not fully testable.

That said, I can't think of an easy way to refactor around this just due to the whole purpose of the method. Here's a stackoverflow that illustrates refactoring around this cop - http://codereview.stackexchange.com/questions/86405/reduce-assignment-branch-condition-size

**Comment by rjdellecese** [comment]
Thanks Dave, these are really helpful comments!

**Comment by rjdellecese** [comment]
@lasley See updates in latest commit--I think this is ready to merge now!

**Comment by lasley** [comment]
Agreed

**Pull request merged**